### PR TITLE
Cleaner implementation

### PR DESCRIPTION
**Pull request by mscalora**
Use all(), map() and reduce() to do the work

### CODE REVIEW

File: lib/promise-hash.js [hunk 1]
```diff
@@ -1,18 +1,13 @@
...
-    for (const promise in iteratable) {
-      iteratable[promise].then(promiseResult => {
-        result[promise] = promiseResult;
-        cursor++;
-
-        if (cursor >= promiseCount) return resolve(result);
-      }).catch(reject);
-    }
+Promise.hash = Promise.hash || function (hashOfPromises) {
+  let keys = Object.keys(hashOfPromises);
```
Comment (mtimofiiv) [inline lib/promise-hash.js]: Can use `const` here, since this variable will not change.

File: lib/promise-hash.js [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
-
-    for (const promise in iteratable) {
-      iteratable[promise].then(promiseResult => {
-        result[promise] = promiseResult;
-        cursor++;
-
-        if (cursor >= promiseCount) return resolve(result);
-      }).catch(reject);
-    }
+Promise.hash = Promise.hash || function (hashOfPromises) {
```
Comment (mtimofiiv) [inline lib/promise-hash.js:3]: Good thinking here, I hadn't considered that there already may be a polyfill in place!

**Comment by mtimofiiv** [comment]
Hi @mscalora! Thanks for the PR! Just one note on it.

File: lib/promise-hash.js (hunk 1 above)
Comment (mscalora) [reply lib/promise-hash.js]: true, i'll update the PR

**Comment by mtimofiiv** [comment]
Thanks, I have merged you in and done a new release. `1.1.0` now holds your changes.

**Pull request merged**